rdt/transformers/numerical.py

```diff
@@ -487,3 +490,160 @@ def _reverse_transform(self, data):
             data = self._univariate.ppf(scipy.stats.norm.cdf(data))
 
         return super()._reverse_transform(data)
+
+
+class BayesGMMTransformer(NumericalTransformer):
+    """Bayesian GMM transformer."""
+
+    STD_MULTIPLIER = 4
+    DETERMINISTIC_TRANSFORM = False
+    DETERMINISTIC_REVERSE = True
+    COMPOSITION_IS_IDENTITY = False
+
+    def __init__(self, dtype=None, nan='mean', null_column=None, max_clusters=10,
+                 weight_threshold=0.005):
+        super().__init__(dtype=dtype, nan=nan, null_column=null_column)
+        self._max_clusters = max_clusters
+        self._weight_threshold = weight_threshold
+        self._number_of_modes = None
+        self._bgm_transformer = None
+        self._valid_component_indicator = None
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        output_types = {
+            'continuous': 'float',
+            'discrete': 'categorical'
+        }
+        if self.null_transformer and self.null_transformer.creates_null_column():
+            output_types['is_null'] = 'float'
+
+        return self._add_prefix(output_types)
+
+    def _fit(self, data):
+        """Fit the transformer to the data.
+
+        Args:
+            data (pandas.Series):
+                Data to fit to.
+        """
+        self._bgm_transformer = BayesianGaussianMixture(
+            n_components=self._max_clusters,
+            weight_concentration_prior_type='dirichlet_process',
+            weight_concentration_prior=0.001,
+            n_init=1
+        )
+
+        super()._fit(data)
+        data = super()._transform(data)
```

Comment:
What is the reason for this? Is it because of the null transformations

Reply:
Yes, the `BayesianGaussianMixture` doesn't accept null values, so we need to call the transform to remove them.

Reply:
Maybe we should just use the `NullTransformer` directly as it might be removed from the `NumericalTransformer` later on

Reply:
Regarding inheriting from the `NumericalTransformer`, we do use the other parameters (like `rounding`) besides the `NullTransformer`. I also believe it makes sense to inherit from it regardless, the `bgmTransformer` only accepts numerical values as input, and so it is a numerical transformer. From an organizational standpoint it should be together with the other numerical transformers.

For the `NullTransformer`, I added it right now because it fails the integration tests (the already existing ones) which require handling on `NaN` values. The change also doesn't seem that complicated, so doing it in the same PR as the when that removes the `NullTransformer` makes more sense to me than removing it now. I'm also not 100% what the code will look like then, so making the changes right now only to have to change it again later could be a problem.
Finally, since I already did it like this, and the code will work the same way either way, I think we should leave as is regardless, and update it together with the `NumericalTransformer` when we get to it (I've spent way too much time on this PR already...).

Reply:
@amontanez24 Also, the `CopulasTransformer` doesn't accept the `rounding`, `min_value`, `max_value` parameters. Do you know why?